src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -28,92 +31,143 @@ public ToolManifestFinder(DirectoryPath probStart, IFileSystem fileSystem = null
-                            errors.Add(string.Format(LocalizableStrings.InPackage, packageId.ToString()) + joined);
-                        }
-                        else
-                        {
-                            result.Add(new ToolManifestPackage(
-                                packageId,
-                                version,
-                                ToolCommandName.Convert(tools.Value.commands)));
-                        }
+            if (deserializedManifest.version == 0)
+            {
+                errors.Add(
+                    LocalizableStrings.ManifestVersion0 +
+                    path.Value);
+            }
+
+            if (deserializedManifest.version > SupportedVersion)
+            {
+                errors.Add(
+                    string.Format(
+                        LocalizableStrings.ManifestVersionHigherThanSupported,
+                        deserializedManifest.version, SupportedVersion) +
+                    path.Value);
+            }
+
+            foreach (KeyValuePair<string, SerializableLocalToolSinglePackage> tools in deserializedManifest.tools)
+            {
+                var packageLevelErrors = new List<string>();
+                var packageIdString = tools.Key;
+                var packageId = new PackageId(packageIdString);
+
+                string versionString = tools.Value.version;
+                NuGetVersion version = null;
+                if (versionString is null)
+                {
+                    packageLevelErrors.Add(LocalizableStrings.ToolMissingVersion);
+                }
+                else
+                {
+                    var versionParseResult = NuGetVersion.TryParse(
+                        versionString, out version);
+                    if (!versionParseResult)
+                    {
+                        packageLevelErrors.Add(string.Format(LocalizableStrings.VersionIsInvalid, versionString));
                     }
+                }
+                NuGetFramework targetFramework = null;
+                var targetFrameworkString = tools.Value.targetFramework;
+                if (!(targetFrameworkString is null))
+                {
+                    targetFramework = NuGetFramework.Parse(
+                        targetFrameworkString);
 
-                    if (errors.Any())
+                    if (targetFramework.IsUnsupported)
                     {
-                        throw new ToolManifestException(LocalizableStrings.InvalidManifestFilePrefix +
-                            string.Join(string.Empty, errors.Select(e => Environment.NewLine + "  " + e)));
+                        packageLevelErrors.Add(
+                            string.Format(LocalizableStrings.TargetFrameworkIsUnsupported,
+                                targetFrameworkString));
                     }
+                }
 
-                    return result;
+                if (tools.Value.commands == null
```

Comment:
what is tools.Value.commands type?

Reply:
I see it is a string[].

Reply:
if the json misses the field, it is null